src/library_webgl.js

```diff
@@ -626,7 +626,14 @@ var LibraryGL = {
       if (Module['preinitializedWebGLContext']) {
         var ctx = Module['preinitializedWebGLContext'];
 #if MAX_WEBGL_VERSION >= 2
-        webGLContextAttributes.majorVersion = (typeof WebGL2RenderingContext != 'undefined' && ctx instanceof WebGL2RenderingContext) ? 2 : 1;
+      if (typeof WebGL2RenderingContext != 'undefined' && ctx instanceof WebGL2RenderingContext) {
+        webGLContextAttributes.majorVersion = 2;
+      } else if (typeof WebGLRenderingContext != 'undefined' && ctx instanceof WebGLRenderingContext) {
+        webGLContextAttributes.majorVersion = 1;
+      } else {
+        // The context may not be an instance of either class; for example, it may be a debug wrapper.
+        webGLContextAttributes.majorVersion = ctx.getParameter(ctx.VERSION).match(/^WebGL (\d+).(\d+)/)[1];
```

Comment:
I would suggest just using this one line in all cases, but for the first draft I kept the shortcut of checking for an exact class in case it's considered important. Let me know if I should leave it in or simplify the code.

Reply:
I tend to agree, just that one line seems to be enough. The others increase code size for no obvious benefit.

Reply:
Done.